api/lib/spree/api/dependencies.rb

```diff
@@ -1,165 +1,122 @@
-        :storefront_checkout_update_service, :storefront_checkout_complete_service, :storefront_checkout_add_store_credit_service,
-        :storefront_checkout_remove_store_credit_service, :storefront_checkout_get_shipping_rates_service,
-        :storefront_cart_compare_line_items_service, :storefront_cart_serializer, :storefront_credit_card_serializer,
-        :storefront_credit_card_finder, :storefront_shipment_serializer, :storefront_payment_method_serializer, :storefront_country_finder,
-        :storefront_country_serializer, :storefront_menu_serializer, :storefront_menu_finder, :storefront_current_order_finder,
-        :storefront_completed_order_finder, :storefront_order_sorter, :storefront_collection_paginator, :storefront_user_serializer,
-        :storefront_products_sorter, :storefront_products_finder, :storefront_product_serializer, :storefront_taxon_serializer,
-        :storefront_taxon_finder, :storefront_find_by_variant_finder, :storefront_cart_update_service, :storefront_cart_associate_service,
-        :storefront_cart_estimate_shipping_rates_service, :storefront_estimated_shipment_serializer,
-        :storefront_store_serializer, :storefront_address_serializer, :storefront_order_serializer,
-        :storefront_account_create_address_service, :storefront_account_update_address_service, :storefront_address_finder,
-        :storefront_account_create_service, :storefront_account_update_service, :storefront_collection_sorter, :error_handler,
-        :storefront_cart_empty_service, :storefront_cart_destroy_service, :storefront_credit_cards_destroy_service, :platform_products_sorter,
-        :storefront_cart_change_currency_service, :storefront_payment_serializer,
-        :storefront_payment_create_service, :storefront_address_create_service, :storefront_address_update_service,
-        :storefront_checkout_select_shipping_method_service,
-
-        :platform_admin_user_serializer, :platform_coupon_handler, :platform_order_update_service,
-        :platform_order_use_store_credit_service, :platform_order_remove_store_credit_service,
-        :platform_order_complete_service, :platform_order_empty_service, :platform_order_destroy_service,
-        :platform_order_next_service, :platform_order_advance_service,
-        :platform_line_item_create_service, :platform_line_item_update_service, :platform_line_item_destroy_service,
-        :platform_order_approve_service, :platform_order_cancel_service,
-        :platform_shipment_change_state_service, :platform_shipment_create_service, :platform_shipment_update_service,
-        :platform_shipment_add_item_service, :platform_shipment_remove_item_service
-      ].freeze
-
-      attr_accessor *INJECTION_POINTS
-
-      def initialize
-        set_storefront_defaults
-        set_platform_defaults
-      end
-
-      private
-
-      def set_storefront_defaults
+      INJECTION_POINTS_WITH_DEFAULTS = {
```

Comment:
Maybe instead of making this a class-level constant, it could be tied to the instance? I.e. maybe doing:
```rb
private

def injection_points_with_defaults
  @injection_points_with_defaults ||= {...} # The dependency hash
end
```

Then we don't have to pass the procs to make it lazy which removes a lot of boilerplate + gets rid of `if responds_to?(:call)` in the helper

Reply:
But then we would need to define attr_accessors somehow. Not sure if we can do that, since technically there's no guarantee that injection_points_with_defaults would be the same for two different instances of the same dependencies class.